### ISSUE TITLE
Forgotten tie to separate vocabulary schema

### ISSUE THREAD

**Issue by leenadR** [OHDSI/WebAPI#171]
It should be possible to use a single schema for the vocabulary apart from cdm/results schemas for the real data. It works in almost all instances (searching for terms, defining sets) and tools (such as achilles), but cohort generation fails in Atlas. 

Cause of the error is a failure in the WebAPI - A full stack trace attached - the initial event is clicking "generate" in Atlas/Cohort/Generation [webapi-cohort-issue.txt](https://github.com/OHDSI/WebAPI/files/750278/webapi-cohort-issue.txt)

If I define just synpuf_cdm.concept as view (create view synpuf_cdm.concept as select * from vocabschema.concept), it works.

**Comment by chrisknoll** [comment]
Right, some of the webAPI scripts don't have the capability for you to use a schema different from the one in the CDM (which I should add: a cdm isn't truly a cdm unless it has the vocabulary tables in it.  The Vocab tables is part of the cdm schema).  to save on space (so that you don't have copies of the vocabulary for each CDM) it's a good approach to create views in each cdm schema that reference out to a central vocabulary.  

However, you need to ensure that all the CDMs that are reference the centralized vocabulary are 'in sync' with each other with respect to when the ETL was performed and which vocabulary was leveraged.  In our internal environments, we have about 12 CDMs which may not all be built on the same version of the vocabulary for a given time (such as 3 of the CDMs get updated quarterly, another semi-anually).  It is possible that some of the CDMs have 1 version of the vocab while others have another. By keeping the vocabulary localized in the CDM, you know that when you look at tables like CONCEPT_ANCESTOR, you're using the correct vocabulary for the given CDM.

That being said, there's been other requests for people to be able to use a central vocab within the WebAPI scripts (by introducing a new token "vocabulary_database_schema") and use that token in all our scripts.  It's a combination of the above concern (people could shoot themselves in the foot if they use it) and just lack of resources to dig into the query templates and make this change that has kept this from happening in a WebAPI release.  If anyone would like to take this on, they should feel welcome to try it!  I'll mark this issue as Help Wanted and keep it open to see if there are any takers.

**Comment by leenadR** [comment]
Thank you for the explanation - I had not thought about the versions. I was tempted, as concept_ids should never change, it would be sensible to import them only once for all cdms. After all, this seems to be the reason that all tools allow for a separate definition of vocabularies (daimons, Achilles...). But you are right, there is more than only the concept_id :)

It would be nice though to complete the separation for the users, which want to do it. Hence, +1 for keeping it open, but with low priority. Perhaps at one day we can submit the required patches.

**Comment by pavgra** [comment]
Generation was resolved by https://github.com/OHDSI/WebAPI/commit/694957d1780284492b7463fd5905f58e9d7ce0c8

Let us know if there are any other places which ignore Vocab daimon.
